src/Ng/Jobs/Catalog2PackageFixupJob.cs

```diff
@@ -0,0 +1,183 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Auth;
+using NuGet.Packaging.Core;
+using NuGet.Services.Configuration;
+using NuGet.Services.Metadata.Catalog;
+
+namespace Ng.Jobs
+{
+    public class Catalog2PackageFixupJob : NgJob
+    {
+        private const int MaximumPackageProcessingAttempts = 5;
+        private static readonly TimeSpan MaximumPackageProcessingTime = TimeSpan.FromMinutes(10);
+        private static readonly TimeSpan DonePollingInterval = TimeSpan.FromMinutes(1);
+
+        private IServiceProvider _serviceProvider;
+
+        public Catalog2PackageFixupJob(ITelemetryService telemetryService, ILoggerFactory loggerFactory)
+            : base(telemetryService, loggerFactory)
+        {
+            ThreadPool.SetMinThreads(MaxDegreeOfParallelism, 4);
+        }
+
+        public override string GetUsage()
+        {
+            return "Usage: ng catalog2packagefixup"
+                   + $"-{Arguments.Source} <catalog>"
+                   + $"-{Arguments.Verify} true/false"
+                   + $"-{Arguments.StorageAccountName} <azure-account>"
+                   + $"-{Arguments.StorageKeyValue} <azure-key> "
+                   + $"-{Arguments.StorageContainer} <azure-container>";
+        }
+
+        protected override void Init(IDictionary<string, string> arguments, CancellationToken cancellationToken)
+        {
+            var source = arguments.GetOrThrow<string>(Arguments.Source);
+            var storageAccount = arguments.GetOrThrow<string>(Arguments.StorageAccountName);
+            var storageKey = arguments.GetOrThrow<string>(Arguments.StorageKeyValue);
+            var storageContainer = arguments.GetOrThrow<string>(Arguments.StorageContainer);
+            var verify = arguments.GetOrDefault(Arguments.Verify, false);
+            var verbose = arguments.GetOrDefault(Arguments.Verbose, false);
+
+            var services = new ServiceCollection();
+
+            services.AddSingleton(TelemetryService);
+            services.AddSingleton(LoggerFactory);
+            services.AddLogging();
+
+            // Prepare the HTTP Client
+            services.AddSingleton(p =>
+            {
+                var assembly = Assembly.GetEntryAssembly();
+                var assemblyName = assembly.GetName().Name;
+                var assemblyVersion = assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? "0.0.0";
+
+                var httpClient = new HttpClient(new WebRequestHandler
+                {
+                    AllowPipelining = true
+                });
+
+                httpClient.DefaultRequestHeaders.Add("User-Agent", $"{assemblyName}/{assemblyVersion}");
```

Comment:
This is really cool. IMO we may want to use this elsewhere as well.

You may want to add a comment in here describing what we're doing, e.g. using the assembly information to set the User Agent.

Reply:
We already use this code! I stole this from NuGet.Jobs 😄

Reply:
Added a comment describing what's happening.